### ISSUE TITLE
Bootstrap 4 alpha 6

### ISSUE THREAD

**Issue by Ks89** [shakacode/bootstrap-loader#235]
Error with bootstrap-loader 2.0.0-beta.18 and the new Bootstrap 4 alpha 6

```
ERROR in ./~/css-loader!./~/postcss-loader!./~/resolve-url-loader!./~/sass-loader?sourceMap!./~/bootstrap-loader/lib/bootstrap.styles.loader.js?{"bootstrapVersion":4,"useFlexbox":true,"extractStyles":false,"styleLoaders":["style","css","postcss","sass"],"styles":["mixins","normalize","print","reboot","type","images","code","grid","tables","forms","buttons","animation","dropdown","button-group","input-group","custom-forms","nav","navbar","card","breadcrumb","pagination","jumbotron","alert","progress","media","list-group","responsive-embed","close","tags","modal","tooltip","popover","carousel","utilities"],"scripts":["alert","button","carousel","collapse","dropdown","modal","popover","scrollspy","tab","tooltip","util"],"configFilePath":"/Users/ks89/git/My-MEAN-website-client/.bootstraprc","bootstrapPath":"/Users/ks89/git/My-MEAN-website-client/~/bootstrap","bootstrapRelPath":"../bootstrap"}!./~/bootstrap-loader/no-op.js
Module build failed: 
undefined
^
      File to import not found or unreadable: ../bootstrap/scss/_animation.
Parent style sheet: stdin
      in /Users/ks89/git/My-MEAN-website-client/node_modules/bootstrap-loader/no-op.js (line 14, column 1)
 @ ./~/style-loader!./~/css-loader!./~/postcss-loader!./~/resolve-url-loader!./~/sass-loader?sourceMap!./~/bootstrap-loader/lib/bootstrap.styles.loader.js?{"bootstrapVersion":4,"useFlexbox":true,"extractStyles":false,"styleLoaders":["style","css","postcss","sass"],"styles":["mixins","normalize","print","reboot","type","images","code","grid","tables","forms","buttons","animation","dropdown","button-group","input-group","custom-forms","nav","navbar","card","breadcrumb","pagination","jumbotron","alert","progress","media","list-group","responsive-embed","close","tags","modal","tooltip","popover","carousel","utilities"],"scripts":["alert","button","carousel","collapse","dropdown","modal","popover","scrollspy","tab","tooltip","util"],"configFilePath":"/Users/ks89/git/My-MEAN-website-client/.bootstraprc","bootstrapPath":"/Users/ks89/git/My-MEAN-website-client/~/bootstrap","bootstrapRelPath":"../bootstrap"}!./~/bootstrap-loader/no-op.js 4:14-1084 13:2-17:4 14:20-1090
 @ ./~/bootstrap-loader/lib/bootstrap.loader.js!./~/bootstrap-loader/no-op.js
 @ ./~/bootstrap-loader/loader.js
 @ ./src/vendor.ts
 @ multi vendor
```

My bootstraprc:
```
---
# Output debugging info
#loglevel: debug

# Major version of Bootstrap: 3 or 4
bootstrapVersion: 4

# If Bootstrap version 4 is used - turn on/off flexbox model
useFlexbox: true

# Webpack loaders, order matters
styleLoaders:
  - style
  - css
  - postcss
  - sass

# Extract styles to stand-alone css file
# Different settings for different environments can be used,
# It depends on value of NODE_ENV environment variable
# This param can also be set in webpack config:
#   entry: 'bootstrap-loader/extractStyles'
# extractStyles: false
env:
  dev:
    extractStyles: false
  prod:
    extractStyles: true

# Customize Bootstrap variables that get imported before the original Bootstrap variables.
# Thus original Bootstrap variables can depend on values from here. All the bootstrap
# variables are configured with !default, and thus, if you define the variable here, then
# that value is used, rather than the default. However, many bootstrap variables are derived
# from other bootstrap variables, and thus, you want to set this up before we load the
# official bootstrap versions.
# For example, _variables.scss contains:
# $input-color: $gray !default;
# This means you can define $input-color before we load _variables.scss
# preBootstrapCustomizations: ./app/styles/bootstrap/pre-customizations.scss

# This gets loaded after bootstrap/variables is loaded and before bootstrap is loaded.
# A good example of this is when you want to override a bootstrap variable to be based
# on the default value of bootstrap. This is pretty specialized case. Thus, you normally
# just override bootrap variables in preBootstrapCustomizations so that derived
# variables will use your definition.
#
# For example, in _variables.scss:
# $input-height: (($font-size-base * $line-height) + ($input-padding-y * 2) + ($border-width * 2)) !default;
# This means that you could define this yourself in preBootstrapCustomizations. Or you can do
# this in bootstrapCustomizations to make the input height 10% bigger than the default calculation.
# Thus you can leverage the default calculations.
# $input-height: $input-height * 1.10;
# bootstrapCustomizations: ./app/styles/bootstrap/customizations.scss

# Import your custom styles here. You have access to all the bootstrap variables. If you require
# your sass files separately, you will not have access to the bootstrap variables, mixins, clases, etc.
# Usually this endpoint-file contains list of @imports of your application styles.
# appStyles: ./app/styles/app.scss

### Bootstrap styles
styles:

  # Mixins
  mixins: true

  # Reset and dependencies
  normalize: true
  print: true

  # Core CSS
  reboot: true
  type: true
  images: true
  code: true
  grid: true
  tables: true
  forms: true
  buttons: true

  # Components
  animation: true
  dropdown: true
  button-group: true
  input-group: true
  custom-forms: true
  nav: true
  navbar: true
  card: true
  breadcrumb: true
  pagination: true
  jumbotron: true
  alert: true
  progress: true
  media: true
  list-group: true
  responsive-embed: true
  close: true
  tags: true

  # Components w/ JavaScript
  modal: true
  tooltip: true
  popover: true
  carousel: true

  # Utility classes
  utilities: true

### Bootstrap scripts
scripts:
  alert: true
  button: true
  carousel: true
  collapse: true
  dropdown: true
  modal: true
  popover: true
  scrollspy: true
  tab: true
  tooltip: true
  util: true

```

**Comment by nicluo** [comment]
> [v4 Alpha 6 ship list](https://github.com/twbs/bootstrap/issues/20939)
> Renamed .tag back to .badge, like v3, to avoid conflicts with common WordPress styles.
> #21327, #21370: Renamed _animation.scss to _transitions.scss for clarity.

1 Change `animation: true` to `transitions: true`
2 Change `tags: true` to `badge: true`

Theses two steps should solve your issue if you're moving from alpha 5 to 6, or for a new project.

**Comment by nicluo** [comment]
Piggybacking on this issue since it's about alpha 6.

@justin808 Will a pull request for updating v4 configuration files and package.json files to `4.0.0-alpha.6` be considered?

**Comment by justin808** [comment]
What do you guys recommend? Is alpha 6 recommended?

**Comment by justin808** [comment]
Is there a way this can be configurable without requiring a new release for every alpha version? In other words, maybe we put a catch in the loader so it doesn't crash and just informs?

**Comment by nicluo** [comment]
Bootstrap documentation is updated for alpha 6, for example, http://v4-alpha.getbootstrap.com/components/tag/ now returns 404. I'd assume new projects will use alpha 6 from now.

It's hard to say what kinds of changes will happen across the alpha releases, however, a deprecated message will probably be way more helpful than `File to import not found or unreadable: ../bootstrap/scss/_animation`, for users thinking about switching up from alpha 5.

I've enumerated the substitutions here, this should provide a starting point. https://github.com/nicluo/bootstrap-loader/commit/edf376a9ac11cbfa818293cd524a202ded62762e

**Comment by nicluo** [comment]
> Flexbox is now on by default, with no fallback! This means we've dropped IE9 support

https://github.com/twbs/bootstrap/commit/eb2e1102be0f4641ee3e5c4e7853360d5a04e3d8

`useFlexbox` option is also no longer required for alpha 6

**Comment by Ks89** [comment]
The good news is that alpha6 [will be the last alpha](http://blog.getbootstrap.com/)

For blog.bootstrap: **As we head to our first beta, we’ll be focused on not adding anything new, ideally making as few breaking changes as possible,**

```
Up next, our first beta
Like you, we’re very much ready for our first beta release. Luckily, we’re in great shape to get there from this alpha. We have the fewest open issues and pulls we’ve had in easily the last 18 months, and the contributions from the community have been outstanding. As we head to our first beta, we’ll be focused on not adding anything new, ideally making as few breaking changes as possible, and emphasizing documentation quality and bug fixes.

We need your help to get there though. Please dive into this latest release and continue to report bugs and submit pull requests as you can. Every bit helps us improve the next release!
```

**Comment by justin808** [comment]
Let me know if you all have a PR on this one. I'm not using bootstrap v4, so I'll depend on all you to help me review and verify before I push a build. I'm ready whenever you all are.

**Comment by IvanKalinin** [comment]
Hi everyone.
How can show a right webpack 2 rc3 configuration with bootstap 4?

Here is my [.bootstraprc](https://gist.github.com/IvanKalinin/92bdc83472bd685768e0ddd23e3712a2)

This is my [webpack.config.js](https://gist.github.com/IvanKalinin/7a12dfca10573ff4e94a0d726dd7d329)

And after all I get this error on npm start:

```Error in ./~/extract-text-webpack-plugin/loader.js?{"omit":1,"remove":true}!./~/style-loader!./~/resolve-url-loader!./~/sass-loader?sourceMap!./~/bootstrap-loader/lib/bootstrap.styles.loader.js?{"bootstrapVersion":4,"extractStyles":true,"styleLoaders":["style-loader","css-loader","postcss-loader","sass-loader"],"styles":["mixins","normalize","print","reboot","type","images","code","grid","tables","forms","buttons","transitions","dropdown","button-group","input-group","custom-forms","nav","navbar","card","breadcrumb","pagination","jumbotron","alert","progress","media","list-group","responsive-embed","close","badge","modal","tooltip","popover","carousel","utilities"],"scripts":false,"configFilePath":"/Users/ivan/Dev/Isolary-Dev/react/.bootstraprc","bootstrapPath":"/Users/ivan/Dev/Isolary-Dev/react/~/bootstrap","bootstrapRelPath":"../bootstrap"}!./~/bootstrap-loader/no-op.js
Module build failed: ModuleBuildError: Module build failed: TypeError: Cannot read property 'path' of undefined 
  @ ./~/bootstrap-loader/lib/bootstrap.loader.js?extractStyles!./~/bootstrap-loader/no-op.js 1:21-1040
  @ ./~/bootstrap-loader/extractStyles.js
  @ multi app

Error in ./~/extract-text-webpack-plugin/loader.js?{"omit":1,"remove":true}!./~/style-loader!./~/resolve-url-loader!./~/sass-loader?sourceMap!./~/bootstrap-loader/lib/bootstrap.styles.loader.js?{"bootstrapVersion":4,"extractStyles":true,"styleLoaders":["style-loader","css-loader","postcss-loader","sass-loader"],"styles":["mixins","normalize","print","reboot","type","images","code","grid","tables","forms","buttons","transitions","dropdown","button-group","input-group","custom-forms","nav","navbar","card","breadcrumb","pagination","jumbotron","alert","progress","media","list-group","responsive-embed","close","badge","modal","tooltip","popover","carousel","utilities"],"scripts":false,"configFilePath":"/Users/ivan/Dev/Isolary-Dev/react/.bootstraprc","bootstrapPath":"/Users/ivan/Dev/Isolary-Dev/react/~/bootstrap","bootstrapRelPath":"../bootstrap"}!./~/bootstrap-loader/no-op.js
Module build failed: ModuleBuildError: Module build failed: TypeError: Cannot read property 'path' of undefined
  at Object.<anonymous> (/Users/ivan/Dev/Isolary-Dev/react/node_modules/async/dist/async.js:2234:31)```

**Comment by IvanKalinin** [comment]
I've fixed my issues.
The problem was in webpack.config.js

How it was:
new webpack.LoaderOptionsPlugin({
...............................................
  options: {
    context: path.resolve(__dirname, '.'),
  },
...............................................
})

**Comment by IvanKalinin** [comment]
I've fixed my issues.
The problem was in webpack.config.js

How it was:
```
new webpack.LoaderOptionsPlugin({
...
  options: {
    ...
    context: path.resolve(__dirname, '.'),
    ...
  },
...
})
```

How should it be
```
new webpack.LoaderOptionsPlugin({
...
  options: {
    ...
    context: path.resolve(__dirname, '.'),
    output: {
      path: 'dist',
    },
    ...
  },
...
```

**Comment by justin808** [comment]
Released 2.1.0
